google/cloud/spanner/row.h

```diff
@@ -183,6 +183,25 @@ class Row {
   std::tuple<Types...>&& get() && { return std::move(values_); }
   std::tuple<Types...> const&& get() const&& { return std::move(values_); }
 
+  /**
+   * Returns a std::array of `Value` objects holding all the items in this row.
+   *
+   * @par Example
+   *
+   * @code
+   * auto row = MakeRow(true, "foo", 42);
+   * std::array<Value, 3> a = row.values();
+   * assert(a[0].get<bool>() == true);
+   * assert(a[1].get<std::string>() == "foo");
+   * assert(a[2].get<std::int64_t>() == 42);
+   * @endcode
+   */
+  std::array<Value, size()> values() const {
```

Comment:
Can we have a `&&` overload for this function?  Otherwise we are always going to copy all the values, and they could be  large (`BYTES`).

Reply:
Done. I added an overload for `values() &&`. I did not add overloads for `&`, because I it should behave the same as `const&`. I also didn't add an overload for `const&&`, because it's a really weird overload anyway, and I think it should behave the same as `const&` too (which it will).

Also, testing move semantics is really hard (not even sure how) with types that we don't own. So I wrote a test that verifies that properly written code will do what it's supposed to do, but I do not verify that the data (i.e., the std::string data) is actually moved.